packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -44,10 +44,18 @@ const headingsCacheKey = node =>
   `transformer-remark-markdown-headings-${
     node.internal.contentDigest
   }-${pluginsCacheStr}-${pathPrefixCacheStr}`
-const tableOfContentsCacheKey = node =>
-  `transformer-remark-markdown-toc-${
+const tableOfContentsCacheKey = function(node, appliedTocOptions) {
+  // define toc options cache string as empty
+  let tocOptionsCacheStr = ``
+  // put applied toc options into toc options cache string
+  Object.keys(appliedTocOptions).map(function(key, index) {
+    tocOptionsCacheStr += key + index
+  })
```

Comment:
You can make generating of `tocOptionsCacheStr` more compact by just using `const tocOptionsCacheStr = JSON.stringify(appliedTocOptions)` - and compact it even more

Reply:
changed in new commit